impl/utility.cxx

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifdef _WIN32
+#include <windows.h>
+#include <dbghelp.h>
+#endif
+#ifndef _MSC_VER
+#include <cxxabi.h>
+#endif
+
+#include <string>
+#include <system_error>
+#include <vector>
+
+#include "substrate/utility"
+
+namespace substrate
+{
+	std::string decode_typename(const char *const mangled_name)
+	{
+#ifndef _MSC_VER
+		int status{};
+		auto * const demangler{ abi::__cxa_demangle(mangled_name, nullptr, nullptr, &status) };
```

Comment:
Please remove the space between the `*` and `const` as in `auto *const`. Likewise after the opening `{` and before the closing `}`